### DISCUSSION TITLE
Have CSSType check the sx prop

### DISCUSSION THREAD

**Discussion by paulzakin** [mantinedev/mantine#2692] [Feature requests]
Currently, if there is incorrect stuff in `sx`, TypeScript will not complain. This seems like a good use case for [CSSType](https://www.npmjs.com/package/csstype)?

Here's an example of what I'm talking about. 

Sandbox: https://codesandbox.io/s/small-moon-3frrkh?file=/src/App.tsx

Snippet:

```tsx
import { MantineProvider, Button } from "@mantine/core";

export default function App() {
  return (
    <MantineProvider withGlobalStyles withNormalizeCSS>
      {/* Correct */}
      <Button sx={{ color: "red" }}>Foo</Button>
      {/* Wrong key */}
      <Button sx={{ colr: "blue" }}>Foo</Button>
      {/* Wrong value */}
      <Button sx={{ color: "yelow" }}>Foo</Button>
    </MantineProvider>
  );
}
```

**Comment by rtivital** [comment]
We already include it. There is no such thing as a wrong key – property can be any string, for example `&::before`. `color` property can have any string value, for example `#e4e` or `rgba(12, 41, 44, 0.5)`.

**Reply by paulzakin** [reply]
Huh - maybe I misread the docs of CSSType - but it looks like it will throw an error on certain things. For example, if color is spelled wrong. 

import type * as CSS from 'csstype';

const style: CSS.Properties = {
  colour: 'white', // Type error on property
  textAlign: 'middle', // Type error on value
};

Maybe I’m not understanding something properly?

**Reply by rtivital** [reply]
You can find CSSType logic here – https://github.com/mantinedev/mantine/blob/dev/src/mantine-styles/src/tss/types/css-object.ts#L1

**Reply by paulzakin** [reply]
Great, I’ll check it out!
